training/src/main/java/com/wolox/training/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.wolox.training.services.impl;
+
+import com.wolox.training.exceptions.BookAlreadyOwnedException;
+import com.wolox.training.exceptions.BookNotFoundException;
+import com.wolox.training.exceptions.UserNotFoundException;
+import com.wolox.training.models.Book;
+import com.wolox.training.models.User;
+import com.wolox.training.repositories.BookRepository;
+import com.wolox.training.repositories.UserRepository;
+import com.wolox.training.services.UserService;
+import java.util.List;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class UserServiceImpl implements UserService {
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @Autowired
+  private BookRepository bookRepository;
+
+  @Override
+  public List<User> getAllUser() {
+    return this.userRepository.findAll();
+  }
+
+  @Override
+  public void saveUser(User userInput) {
+
+    log.info("saving book..., {}", userInput);
+    this.userRepository.save(userInput);
+  }
+
+  @Override
+  public User findUserById(Long id) throws UserNotFoundException {
+    Optional<User> user = this.userRepository.findById(id);
+    if (user.isPresent()) {
+      return user.get();
+    } else {
+      throw new UserNotFoundException("User not found", 404);
+    }
```

Comment:
I think the code on this method could be replaced with this:
```
this.userRepository.findById(id)
   .orElseThrow(new UserNotFoundException("User not found", 404)); 
```
This means that if the value exists, it will be returned. But if not, then an exception will be thrown